runtime/mainnet/src/config/xcm.rs

```diff
@@ -215,56 +258,867 @@ impl cumulus_pallet_xcm::Config for Runtime {
+		#[test]
+		fn paused_query_handler_uses_xcmp_queue() {
+			assert_eq!(
+				TypeId::of::<<Runtime as pallet_message_queue::Config>::QueuePausedQuery>(),
+				TypeId::of::<NarrowOriginToSibling<XcmpQueue>>()
+			);
+		}
+
+		#[test]
+		fn limits_service_weight() {
+			assert_eq!(
+				<<Runtime as pallet_message_queue::Config>::ServiceWeight as Get<Weight>>::get(),
+				Perbill::from_percent(35) * RuntimeBlockWeights::get().max_block
+			);
+		}
+
+		#[test]
+		fn uses_u32_page_size() {
+			assert_eq!(
+				TypeId::of::<<Runtime as pallet_message_queue::Config>::Size>(),
+				TypeId::of::<u32>()
+			);
+		}
+
+		#[test]
+		fn does_not_use_default_weights() {
+			assert_ne!(
+				TypeId::of::<<Runtime as pallet_message_queue::Config>::WeightInfo>(),
+				TypeId::of::<()>(),
+			);
+		}
 	}
 
-	#[test]
-	fn relay_as_relay_asset_reserve_fails() {
-		let relay_asset = Asset::from((AssetId::from(Parent), Fungibility::from(100u128)));
-		assert!(!TrustedReserves::contains(&relay_asset, &Parent.into()));
+	mod xcm_configuration {
+		use super::*;
+
+		#[test]
+		fn location_to_account_id_matches_configuration() {
+			assert_eq!(
+				TypeId::of::<LocationToAccountId>(),
+				TypeId::of::<(
+					ParentIsPreset<AccountId>,
+					SiblingParachainConvertsVia<Sibling, AccountId>,
+					AccountId32Aliases<RelayNetwork, AccountId>,
+				)>()
+			);
+		}
+
+		#[test]
+		fn local_asset_transactor_matches_configuration() {
+			assert_eq!(
+				TypeId::of::<LocalAssetTransactor>(),
+				TypeId::of::<
+					FungibleAdapter<
+						Balances,
+						IsConcrete<RelayLocation>,
+						LocationToAccountId,
+						AccountId,
+						(),
+					>,
+				>()
+			);
+		}
+
+		#[test]
+		fn xcm_origin_to_transact_dispatch_origin_matches_configuration() {
+			assert_eq!(
+				TypeId::of::<XcmOriginToTransactDispatchOrigin>(),
+				TypeId::of::<(
+					SovereignSignedViaLocation<LocationToAccountId, RuntimeOrigin>,
+					RelayChainAsNative<RelayChainOrigin, RuntimeOrigin>,
+					SiblingParachainAsNative<cumulus_pallet_xcm::Origin, RuntimeOrigin>,
+					SignedAccountId32AsNative<RelayNetwork, RuntimeOrigin>,
+					XcmPassthrough<RuntimeOrigin>,
+				)>()
+			);
+		}
+
+		#[test]
+		fn barrier_configuration() {
+			assert_eq!(
+				TypeId::of::<Barrier>(),
+				TypeId::of::<
+					TrailingSetTopicAsId<(
+						TakeWeightCredit,
+						AllowKnownQueryResponses<PolkadotXcm>,
+						WithComputedOrigin<
+							(
+								AllowTopLevelPaidExecutionFrom<Everything>,
+								AllowSubscriptionsFrom<ParentRelayOrSiblingParachains>,
+							),
+							UniversalLocation,
+							ConstU32<8>,
+						>,
+					)>,
+				>()
+			);
+		}
 	}
 
-	// Decline native asset from another parachain.
-	#[test]
-	fn decline_sibling_native_assets() {
-		let chain_x = Location::new(1, [Parachain(4242)]);
-		let chain_x_asset =
-			Asset::from((AssetId::from(chain_x.clone()), Fungibility::from(100u128)));
-		assert!(!TrustedReserves::contains(&chain_x_asset, &chain_x));
+	mod xcm_executor_configuration {
```

Comment:
We are in the file that represents the configuration of xcm so I find it unnecessary to add
```suggestion
	mod xcm_executor {
```

Reply:
Naming will conflict with `<XcmConfig as xcm_executor::Config>`